src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ResultPropertiesTests.cs

```diff
@@ -365,6 +365,48 @@ void Test()
         }
 
         [Fact]
+        public void DeconstructionDeclaration()
```

Comment:
Please move test to `DeclarationTests` and verify the IL of the generated method. The IL should contain two calls to `IntrinsicMethods.CreateVariable`. See `DeclarationTests.Declarations` for an example.